markets/bfp-market/contracts/modules/PerpAccountModule.sol

```diff
@@ -118,4 +136,128 @@ contract PerpAccountModule is IPerpAccountModule {
+        PerpMarket.Data storage market = PerpMarket.exists(marketId);
+        PerpMarketConfiguration.Data storage marketConfig = PerpMarketConfiguration.load(marketId);
+        Margin.GlobalData storage globalMarginConfig = Margin.load();
+        Margin.Data storage fromAccountMargin = Margin.load(fromId, marketId);
+
+        Runtime_mergeAccounts memory runtime;
+
+        runtime.supportedSynthMarketIdsLength = globalMarginConfig.supportedSynthMarketIds.length;
+
+        for (uint256 i = 0; i < runtime.supportedSynthMarketIdsLength; ) {
+            runtime.synthMarketIdForLoop = globalMarginConfig.supportedSynthMarketIds[i];
+            runtime.fromAccountCollateralForLoop = fromAccountMargin.collaterals[
+                runtime.synthMarketIdForLoop
+            ];
+            if (runtime.fromAccountCollateralForLoop > 0) {
+                if (
+                    globalMarginConfig.supported[runtime.synthMarketIdForLoop].oracleNodeId !=
+                    marketConfig.oracleNodeId
+                ) {
+                    // If the from account have collateral >0 with different oracle node id then the market, revert.
+                    revert ErrorUtil.OracleNodeMismatch();
+                }
+                // This is the collateral we're intrested in and it has correct oracle node id.
+                runtime.synthMarketId = runtime.synthMarketIdForLoop;
+                runtime.fromAccountCollateral = runtime.fromAccountCollateralForLoop;
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        Margin.Data storage toAccountMargin = Margin.load(toId, marketId);
+        Position.Data storage fromPosition = market.positions[fromId];
+        Position.Data storage toPosition = market.positions[toId];
+
+        if (fromPosition.entryTime != block.timestamp) {
+            revert ErrorUtil.PositionTooOld();
+        }
+        if (market.orders[toId].sizeDelta != 0) {
+            // We only have to check "toAccount" as its impossible for fromAccount to have an order.
+            revert ErrorUtil.OrderFound();
+        }
+
+        runtime.oraclePrice = market.getOraclePrice();
+
+        realizePosition(
+            market,
+            toPosition,
+            Margin.getMarginUsd(toId, market, runtime.oraclePrice),
+            toAccountMargin,
+            marketConfig,
+            runtime.oraclePrice
+        );
+
+        // Move collateral.
+        toAccountMargin.collaterals[runtime.synthMarketId] += runtime.fromAccountCollateral;
+        fromAccountMargin.collaterals[runtime.synthMarketId] = 0;
+
+        // Update toAccount's postion with data from the fromAccount's position.
+        Position.Data memory newPosition = Position.Data(
+            toPosition.size + fromPosition.size,
+            block.timestamp,
+            market.currentFundingAccruedComputed,
+            // Since utilization wont be recomputed here we need to manually add the unrecorded utilization.
```

Comment:
🤔

I thought during settlement funding and utilisation is recomputed and with the `block.timestamp == fromPosition.timestamp` check, wouldn't `getUnrecordedUtilization` just be zero?

Reply:
Yea true.. Missed removing this as I changes to the same block model